docs/core/tutorials/unloadability-howto.md

```diff
@@ -0,0 +1,362 @@
+# Using and debugging unloadability in .NET Core
+An ability to load and later unload a set of assemblies is one of the features that was missing until now in .NET Core. In .NET Framework, custom app domains are used for this purpose. But since .NET Core supports only a single default app domain, how can we achieve that goal in .NET Core?
+
+The .NET Core 3.0 supports unloadability through <xref:System.Runtime.Loader.AssemblyLoadContext>. You can load a set of assemblies into a collectible `AssemblyLoadContext`, execute methods in them or just inspect them using reflection, and finally unload the `AssemblyLoadContext`. That unloads the assemblies loaded into that `AssemblyLoadContext`.
+
+There is one noteworthy difference between the unloading using `AssemblyLoadContext` and using AppDomains. With AppDomains, the unloading is forced. At the unload time, all threads running in the target AppDomain are aborted, managed COM objects created in the target AppDomain are destroyed, etc. With `AssemblyLoadContext`, the unload is "cooperative". Calling the <xref:System.Runtime.Loader.AssemblyLoadContext.Unload%2A?displayProperty=nameWithType> method just initiates the unloading. It does not complete until:
+- No threads have code from the assemblies loaded into the `AssemblyLoadContext` on their call stacks.
```

Comment:
Following the sentence as a whole reads a bit weird...

It doesn't complete until no threads have code from the assemblies loaded into the `AssemblyLoadContext` on their call stacks.

Reply:
I've reworded this part to make it easier to understand.